src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.cs

```diff
@@ -116,20 +118,23 @@ private static void Json(PipeWriter pipeWriter)
 
             // Content-Length header
             writer.Write(_headerContentLength);
-            var jsonPayload = JsonSerializer.SerializeToUtf8Bytes(new JsonMessage { message = "Hello, World!" }, SerializerOptions);
-            writer.WriteNumeric((uint)jsonPayload.Length);
+
+            writer.WriteNumeric((uint)JsonPayload.Length);
 
             // End of headers
             writer.Write(_eoh);
+            writer.Commit();
 
             // Body
-            writer.Write(jsonPayload);
-            writer.Commit();
+            using (var utf8jsonWriter = new Utf8JsonWriter(writer.Output))
```

Comment:
the main difference for using `Utf8JsonWriter` is that we don't allocate an array of bytes